File: Task/Erd-s-Nicolas-numbers/C-sharp/erd-s-nicolas-numbers.cs
using System;

class ErdosNicolasNumbers
{
    static void Main(string[] args)
    {
        const int limit = 100_000_000;

        int[] divisorSum = new int[limit + 1];
        int[] divisorCount = new int[limit + 1];
        for (int i = 0; i <= limit; i++)
        {
            divisorSum[i] = 1;
            divisorCount[i] = 1;
        }

        for (int index = 2; index <= limit / 2; index++)
        {
            for (int number = 2 * index; number <= limit; number += index)
            {
                if (divisorSum[number] == number)
                {
                    Console.WriteLine($"{number,8} equals the sum of its first {divisorCount[number],3} divisors");
                }

                divisorSum[number] += index;
                divisorCount[number]++;
            }
        }
    }
}

File: Task/Erd-s-Nicolas-numbers/PascalABC.NET/erd-s-nicolas-numbers.pas
const
  max_number = 100_000_000;

begin
  var dsum := arrfill(max_number + 1, 1);
  var dcount := arrfill(max_number + 1, 1);
  for var i := 2 to max_number do
    for var j := i + i to max_number step i do
    begin
      if dsum[j] = j then
        writeln(j:8, ' equals the sum of its first ', dcount[j], ' divisors');
      dsum[j] += i;
      dcount[j] += 1;
    end;
end.
